model/src/main/java/com/flipkart/flux/domain/Context.java

```diff
@@ -32,7 +33,7 @@
     /** Identifier for the Context*/
     protected String contextId;
     /** A dependency graph created across States - holds information on possible next state transitions for a State, keyed by the Event FQN*/
-    protected Map<String,List<State<T>>> stateToEventDependencyGraph;
+    protected Map<String, Map<String, State<T>>> stateToEventDependencyGraph;
```

Comment:
My understanding is stateToEventDependencyGraph must represent DAG for the StateMachine.
Now, from a State its Task can raise *several* Events. And each Event will land up on another *single* State.
So its a Map(PivotState, Map(EventFqn, TransitionState))